lib/ai_bot/tools/tool.rb

```diff
@@ -77,8 +77,34 @@ def standalone?
 
         protected
 
-        def send_http_request(url, headers: {}, authenticate_github: false)
-          uri = URI(url)
+        def send_http_request(url, headers: {}, authenticate_github: false, follow_redirects: false)
+          raise "Expecting caller to use a block" if !block_given?
+
+          uri = nil
+
+          if follow_redirects
+            url = UrlHelper.normalized_encode(url)
+            URI.parse(url) # ensure url parses, will raise if not
```

Comment:
Why raise here for `URI.parse` and `rescue StandardError` below if not following redirects?

Reply:
hmmm I am not sure maybe I just return null and then it give no response back to caller and it gets an unknown to the llm

Reply:
fixed